Library/Homebrew/test/exceptions_spec.rb

```diff
@@ -0,0 +1,180 @@
+require "exceptions"
+
+describe MultipleVersionsInstalledError do
+  subject { described_class.new("foo") }
+  its(:to_s) { is_expected.to eq("foo has multiple installed versions") }
+end
+
+describe NoSuchKegError do
+  subject { described_class.new("foo") }
+  its(:to_s) { is_expected.to eq("No such keg: #{HOMEBREW_CELLAR}/foo") }
+end
+
+describe FormulaValidationError do
+  subject { described_class.new("foo", "sha257", "magic") }
+  its(:to_s) { is_expected.to eq(%q(invalid attribute for formula 'foo': sha257 ("magic"))) }
```

Comment:
Maybe worth splitting some of these lines that are quite a bit over 80 characters.